libr/io/p/io_fd.c

```diff
@@ -0,0 +1,99 @@
+/* radare - LGPL - Copyright 2020 - pancake */
+
+#include "r_io.h"
+#include "r_lib.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <r_cons.h>
+#include <sys/types.h>
+
+typedef struct {
+	int fd;
+	ut32 size;
+} RIOMalloc;
+
+static int __write(RIO *io, RIODesc *desc, const ut8 *buf, int count) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		return write (mal->fd, buf, count);
+	}
+	return -1;
+}
+
+static bool __resize(RIO *io, RIODesc *desc, ut64 count) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		return ftruncate (mal->fd, count) == 0;
+	}
+	return false;
+}
+
+static int __read(RIO *io, RIODesc *desc, ut8 *buf, int count) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		r_cons_break_push (NULL, NULL);
+		int res = read (mal->fd, buf, count);
+		r_cons_break_pop ();
+		return res;
+	}
+	return -1;
+}
+
+static int __close(RIODesc *desc) {
+	// dont close, could be problematic in self://
```

Comment:
I think we should at least free the RIODesc allocated in `__open`

Reply:
this is done by r_io_close, but by the plugin callback. im freeing the ->data inside now